digraph "unitGraph" {
    "Block 0:
[preds: ] [succs: 1 2 ]
r10 := @parameter0: java.lang.String[];
$r0 = new java.lang.Throwable;
specialinvoke $r0.<java.lang.Throwable: void <init>(java.lang.String)>(\"because\");
r1 = $r0;
$r2 = new java.lang.Throwable;
specialinvoke $r2.<java.lang.Throwable: void <init>(java.lang.Throwable)>(r1);
r3 = $r2;
$r4 = new java.lang.TypeNotPresentException;
specialinvoke $r4.<java.lang.TypeNotPresentException: void <init>(java.lang.String,java.lang.Throwable)>(\"test\", r3);
r5 = $r4;
"
    "Block 1:
[preds: 0 ] [succs: 2 ]
throw r5;
"
    "Block 2:
[preds: 0 1 ] [succs: 3 4 ]
$r6 := @caughtexception;
r7 = $r6;
$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>();
if r3 == $r8 goto return;
"
    "Block 3:
[preds: 2 ] [succs: ]
$r9 = new java.lang.RuntimeException;
specialinvoke $r9.<java.lang.RuntimeException: void <init>(java.lang.String)>(\"Unexpected value of cause.\");
throw $r9;
"
    "Block 4:
[preds: 2 ] [succs: ]
return;
"
    "Block 0:
[preds: ] [succs: 1 2 ]
r10 := @parameter0: java.lang.String[];
$r0 = new java.lang.Throwable;
specialinvoke $r0.<java.lang.Throwable: void <init>(java.lang.String)>(\"because\");
r1 = $r0;
$r2 = new java.lang.Throwable;
specialinvoke $r2.<java.lang.Throwable: void <init>(java.lang.Throwable)>(r1);
r3 = $r2;
$r4 = new java.lang.TypeNotPresentException;
specialinvoke $r4.<java.lang.TypeNotPresentException: void <init>(java.lang.String,java.lang.Throwable)>(\"test\", r3);
r5 = $r4;
"->"Block 1:
[preds: 0 ] [succs: 2 ]
throw r5;
";
    "Block 0:
[preds: ] [succs: 1 2 ]
r10 := @parameter0: java.lang.String[];
$r0 = new java.lang.Throwable;
specialinvoke $r0.<java.lang.Throwable: void <init>(java.lang.String)>(\"because\");
r1 = $r0;
$r2 = new java.lang.Throwable;
specialinvoke $r2.<java.lang.Throwable: void <init>(java.lang.Throwable)>(r1);
r3 = $r2;
$r4 = new java.lang.TypeNotPresentException;
specialinvoke $r4.<java.lang.TypeNotPresentException: void <init>(java.lang.String,java.lang.Throwable)>(\"test\", r3);
r5 = $r4;
"->"Block 2:
[preds: 0 1 ] [succs: 3 4 ]
$r6 := @caughtexception;
r7 = $r6;
$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>();
if r3 == $r8 goto return;
";
    "Block 1:
[preds: 0 ] [succs: 2 ]
throw r5;
"->"Block 2:
[preds: 0 1 ] [succs: 3 4 ]
$r6 := @caughtexception;
r7 = $r6;
$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>();
if r3 == $r8 goto return;
";
    "Block 2:
[preds: 0 1 ] [succs: 3 4 ]
$r6 := @caughtexception;
r7 = $r6;
$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>();
if r3 == $r8 goto return;
"->"Block 3:
[preds: 2 ] [succs: ]
$r9 = new java.lang.RuntimeException;
specialinvoke $r9.<java.lang.RuntimeException: void <init>(java.lang.String)>(\"Unexpected value of cause.\");
throw $r9;
";
    "Block 2:
[preds: 0 1 ] [succs: 3 4 ]
$r6 := @caughtexception;
r7 = $r6;
$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>();
if r3 == $r8 goto return;
"->"Block 4:
[preds: 2 ] [succs: ]
return;
";
}
